res/values/strings.xml

```diff
@@ -1,13 +1,13 @@
 <?xml version="1.0" encoding="utf-8"?>
 <resources>
-    <string name="mtm_accept_cert">Accept Unknown Certificate?</string>
+    <string name="mtm_security_risk">Warning: Potential Security Risk</string>
     <string name="mtm_trust_anchor">The server certificate is not signed by a known Certificate Authority.</string>
     <string name="mtm_cert_expired">The server certificate is expired.</string>
+    <string name="mtm_hostname_mismatch">Server could not authenticate as \&quot;%s\&quot;.</string>
+    <string name="mtm_trust_certificte">Do you trust this certificte? It will be used to verify all listed names.</string>
     <string name="mtm_accept_servername">Accept Mismatching Server Name?</string>
-    <string name="mtm_hostname_mismatch">Server could not authenticate as \&quot;%s\&quot;. The certificate is only valid for:</string>
-
-    <string name="mtm_connect_anyway">Do you want to connect anyway?</string>
     <string name="mtm_cert_details">Certificate details:</string>
+    <string name="mtm_sans">Subject alt names:</string>
```

Comment:
"The certificate is valid for:" is a much better non-technical title than SANs, however it only makes sense in the hostname-mismatch case. I'd like to have some sort of better rendering of the certificate chain, as I'm using the same code for all certificates, that I repeatedly call for the chain.

Maybe something like this would work - fixed string "Server certificate:", then certDetails(), then as many "Signed by:" + certDetails() pairs as needed:

> Server certificate:
> [2] *.badssl.com
> [2] badssl.com
> 2019-10-10 - 2021-10-09
> SHA...
> SHA...
> Signed by: [Issuer DN]
>
> Signed by:
> CN=BadSSL Untrusted...
> 2016-07-07 - 2036-07-02
> SHA...

But of course this is only a quirk because we do not have collapsible views or something like a tree-view as used in the browsers.

Reply:
I think that in your case it's not very clear what the domains are.

Reply:
_The certificate is valid for:_ would be very misleading. I think that _Subject alt names:_ is the best wording here. What would even non-technical users be doing there?

Reply:
Entire chain will be shown even when there is no relation between certs:
![Screenshot_2021-04-28_153652](https://user-images.githubusercontent.com/8530546/116413132-a0d65200-a837-11eb-940e-2df6aeb24738.png)
![Screenshot_2021-04-28_153657](https://user-images.githubusercontent.com/8530546/116413139-a16ee880-a837-11eb-92af-0f962f8d1440.png)